tests/api/auth/backend/test_basic_auth.py

```diff
@@ -0,0 +1,106 @@
+from parameterized import parameterized
+
+from airflow.www.app import create_app
+from tests.test_utils.config import conf_vars
+from tests.test_utils.db import clear_db_pools
+
+
+class TestBasicAuth(unittest.TestCase):
+    def setUp(self) -> None:
+        with conf_vars(
+            {("api", "auth_backend"): "airflow.api.auth.backend.basic_auth"}
+        ):
+            self.app = create_app(testing=True)
+
+        self.appbuilder = self.app.appbuilder  # pylint: disable=no-member
+        role_admin = self.appbuilder.sm.find_role("Admin")
+        tester = self.appbuilder.sm.find_user(username="test")
+        if not tester:
+            self.appbuilder.sm.add_user(
+                username="test",
+                first_name="test",
+                last_name="test",
+                email="test@fab.org",
+                role=role_admin,
+                password="test",
+            )
+
+    def test_success(self):
+        token = "Basic " + b64encode(b"test:test").decode()
+        clear_db_pools()
+
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert current_user.email == "test@fab.org"
+
+        assert response.status_code == 200
+        assert response.json == {
+            "pools": [
+                {
+                    "name": "default_pool",
+                    "slots": 128,
+                    "occupied_slots": 0,
+                    "running_slots": 0,
+                    "queued_slots": 0,
+                    "open_slots": 128,
+                },
+            ],
+            "total_entries": 1,
+        }
+
+    @parameterized.expand([
+        ("basic",),
+        ("basic ",),
+        ("bearer",),
+        ("test:test",),
+        (b64encode(b"test:test").decode(),),
+        ("bearer ",),
+        ("basic: ",),
+        ("basic 123",),
+    ])
+    def test_malformed_headers(self, token):
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert response.status_code == 401
+            assert response.headers["WWW-Authenticate"] == "Basic"
+            assert response.data == b"Unauthorized"
+
+    @parameterized.expand([
+        ("basic " + b64encode(b"test").decode(),),
+        ("basic " + b64encode(b"test:").decode(),),
+        ("basic " + b64encode(b"test:123").decode(),),
+        ("basic " + b64encode(b"test test").decode(),),
+    ])
+    def test_invalid_auth_header(self, token):
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert response.status_code == 401
+            assert response.headers["WWW-Authenticate"] == "Basic"
+            assert response.data == b"Unauthorized"
```

Comment:
This answer should be in line with RFC-7807.
https://airflow.readthedocs.io/en/latest/stable-rest-api/redoc.html

Reply:
https://github.com/apache/airflow/blob/master/airflow/api/client/json_client.py#L35-L42
https://github.com/apache/airflow/blob/master/airflow/api/auth/backend/deny_all.py#L41
https://github.com/apache/airflow/blob/master/airflow/api/auth/backend/kerberos_auth.py#L106
https://github.com/apache/airflow/blob/master/airflow/api/auth/backend/kerberos_auth.py#L102

Reply:
updated, for experimental api, the response is set to the same as other auth backends.